### DISCUSSION TITLE
Workflow triggering on PR synchronize

### DISCUSSION THREAD

**Discussion by jurgenvanwestreenen** [community/community#34295] [Actions]
Hi all,

Maybe it's a bug, maybe I'm doing it wrong, maybe it works as designed, but I noticed the following with a workflow using below syntax:

```
on:
    pull_request:
    types: [opened, synchronize]
    branches: [ develop ]
    paths:
      - 'force-app/**'
```

When creating a pull request that contains changes in the force-app folder, it triggers nicely.
However when I do a subsequent commit to that same PR that only contains changes NOT in the force-app folder, it still triggers the workflow.

How does that work? Does it look at the entire PR (not just the latest commit) when evaluating?

What I want to achieve is that commits that don't match the paths argument should **not** trigger the workflow, regardless of any previous commits that **did** trigger the workflow.

**UPDATE:** It actually seems to look at the entire history of commits, because when we tried to merge the changes into the master branch, which shouldn't trigger the workflow, it still performed the check.
Bottom line is, I just want to have a workflow that re-evaluates the criteria for each commit, but solely based on the contents of THAT commit. If anyone has any ideas on how accomplish that, I'm all ears.

**Comment by marcobiglesias** [comment]
Hi! I'm facing the same issue here. @jurgenvanwestreenen were you able to figure out what the problem was?
